addons/web/static/tests/fields/basic_fields_tests.js

```diff
@@ -2937,6 +2937,38 @@ QUnit.module('basic_fields', {
         form.destroy();
     });
 
+    QUnit.test('field date should select its content onclick when there is one', function (assert) {
+        assert.expect(2);
+        var done = assert.async();
+
+        var form = createView({
+            View: FormView,
+            model: 'partner',
+            data: this.data,
+            arch:'<form string="Partners">' +
+                    '<field name="display_name" /> ' + // Do not focus on the date field right away
+                    '<field name="date" />' +
+                '</form>',
+            res_id: 1,
+            viewOptions: {
+                mode: 'edit',
+            },
+        });
+
+        form.$().on({
```

Comment:
form.$()? I'm not even sure of what this is.

Reply:
@aab-odoo 
it just returns this.$el
https://github.com/odoo/odoo/blob/11.0/addons/web/static/src/js/core/widget.js#L343